static_frame/core/type_blocks.py

```diff
@@ -4162,6 +4165,29 @@ def fill_missing_by_values(self,
                         )
                 )
 
+    def iter_shallow_keys(self: "TypeBlocks") -> tp.Iterator[tp.Hashable]:
```

Comment:
@flexatone What do you think about this new approach? It passes all tests and I think it is a good performance improvement

Reply:
This is on the right track, I am just not sure if "shallow_keys" is the right name for this, maybe `iter_block_signature` is better? 

Note that you comparing here most of the `__array_interface__` (from which `mloc` gets is values):

```
>>> a1 = np.array([3,4,5])                                                  
>>> a1                                                                      
array([3, 4, 5])
>>> a1.__array_interface__                                                  
{'data': (94745498307712, False),
 'strides': None,
 'descr': [('', '<i8')],
 'typestr': '<i8',
 'shape': (3,),
 'version': 3}
```

Note that dtype (as typestr) is included there but not in your comparison. Are we confident we do not need to compare dtype? (Is it possible for the same block of memory to be used in two arrays with different dtypes? I do not think so...)